src/main/java/org/jabref/logic/exporter/TemplateExporter.java

```diff
@@ -81,7 +97,11 @@ public TemplateExporter(String displayName, String consoleName, String lfFileNam
     public TemplateExporter(String displayName, String consoleName, String lfFileName, String directory, FileType extension,
             LayoutFormatterPreferences layoutPreferences, SavePreferences savePreferences) {
         super(consoleName, displayName, extension);
-        this.lfFileName = Objects.requireNonNull(lfFileName);
+        if (Objects.requireNonNull(lfFileName).endsWith(".layout")) {
```

Comment:
Maybe extract `".layout"` to a static constant?

Reply:
Eclipse helps you with refactoring. Choose refactor, extract to constant and then you have the option to replace all occurrences with the constant

Reply:
I added a few constants, since `".layout"` was not the online constant string used.